app/src/main/java/ghcharactertracker/com/ghcharactertracker/CharacterActivity.java

```diff
@@ -0,0 +1,304 @@
+
+        dbHandler.updateScenario(newScen);
+        character.setCurrentScenario(newScen);
+
+        updateUI();
+    }
+
+    private void saveIncompleteScenario(Scenario scenario) {
+        character.setCurrentScenario(scenario);
+        dbHandler.updateScenario(scenario);
+    }
+
+    public void newScenario(View v) {
+        /*
+        if (character.getCurrentScenario() == null) {
+            Scenario newScen = new Scenario(character.getMaxHealth(), 0, 0);
+            character.setCurrentScenario(newScen);
+
+            dbHandler.addScenario(newScen);
+            dbHandler.updateCharacter(character);
+
+            startScenarioActivity(newScen);
+        }
+        */
+
+        startScenarioActivity(character.getCurrentScenario());
+    }
+
+    private void startScenarioActivity(Scenario scen) {
+        Intent newScenarioIntent = new Intent(this, ScenarioActivity.class);
+        newScenarioIntent.putExtra(ScenarioActivity.SCENARIO, scen);
+        startActivityForResult(newScenarioIntent, SCENARIO_REQUEST_CODE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch(requestCode) {
+            case SCENARIO_REQUEST_CODE:
+                if (resultCode == Activity.RESULT_OK) { unpackScenarioData(data); };
+                break;
+        }
+    }
+
+    private void addSpinnerWatcher() {
+        classNameSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
+                updateModelClassName(ClassName.values()[position]);
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {}
+        });
+    }
+
+    private void addTextWatchers() {
+        playerName.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (editable.length() > 0) {
+                    updateModelPlayerName(editable.toString());
+                }
+            }
+        });
+
+        playerLevel.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (editable.length() > 0) {
+                    updateModelPlayerLevel(Integer.parseInt(editable.toString()));
```

Comment:
What should happen here when your user finishes deleting the text? Should something trigger?

Same question for the next `afterTextChanged` blocks